src/files.rs

```diff
@@ -234,23 +235,57 @@ pub fn guess_content(bytes: &[u8]) -> ProbableFileKind {
 }
 
 /// All the files in `dir`, including subdirectories.
-fn relative_file_paths_in_dir(dir: &Path) -> Vec<PathBuf> {
+fn relative_file_paths_in_dir(dir: &Path, ignored_dirs: &Option<Vec<PathBuf>>) -> Vec<PathBuf> {
     WalkDir::new(dir)
         .into_iter()
         .filter_map(Result::ok)
         .map(|entry| entry.into_path())
         .filter(|path| !path.is_dir())
+        .filter(|path| !is_ignored(ignored_dirs, path))
         .map(|path| path.strip_prefix(dir).unwrap().to_path_buf())
         .collect()
 }
 
+fn is_ignored(ignored_dirs: &Option<Vec<PathBuf>>, path: &PathBuf) -> bool {
```

Comment:
FWIW it's generally better to take a slice argument in Rust unless you're expecting to push or pop items. In this case you could use `Option<&[PathBuf]>`.